barretenberg/cpp/src/barretenberg/vm/avm_trace/avm_mem_trace.hpp

```diff
@@ -64,6 +66,13 @@ class AvmMemTraceBuilder {
         FF val{};
     };
 
+    // Enum to define which subtable triggered and manages (via perm) the memory operation.
```

Comment:
This is kind of awkward for now, but we need to tag the entry in the mem trace with different selectors based on which subtable is going to performing the PERM check.

`MAIN` is the default value for now, and it's done for backward compatibility